homeassistant/components/autarco/sensor.py

```diff
@@ -25,6 +25,77 @@
 from .coordinator import AutarcoDataUpdateCoordinator
 
 
+@dataclass(frozen=True, kw_only=True)
+class AutarcoBatterySensorEntityDescription(SensorEntityDescription):
+    """Describes an Autarco sensor entity."""
+
+    value_fn: Callable[[Battery], StateType]
+
+
+SENSORS_BATTERY: tuple[AutarcoBatterySensorEntityDescription, ...] = (
```

Comment:
Do these need state classes?

Reply:
Maybe not with state_of_charge, since it only gives the battery percentage.